### DISCUSSION TITLE
How to view error logs (without docker)

### DISCUSSION THREAD

**Discussion by david02871** [open-webui/open-webui#3756] [Q&A]
I installed open web ui via pip and I'm running it with `export WEBUI_AUTH=false && export GLOBAL_LOG_LEVEL=DEBUG && open-webui serve`. I'm trying to debug my custom Tools and Functions but the output is flooded with socket.io requests. What is the best way to view the error logs?

```
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3Jv_&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3JwW&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3Jwp&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3Jx7&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3JxZ&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3Jxs&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
DEBUG:main:Commit session after request
INFO:     127.0.0.1:52309 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P2S3Jy8&sid=dNBoZitMfpAIz77YAAAU HTTP/1.1" 200 OK
...
```

**Comment by david02871** [comment]
Found a solution to my own question although there may be a better way to control the logging level.
`export WEBUI_AUTH=false && open-webui serve | grep --line-buffered -v -E "INFO|DEBUG"`

**Reply by silentoplayz** [reply]
I suppose this may help as well: https://docs.openwebui.com/getting-started/logging

**Reply by JulesJulius** [reply]
> I suppose this may help as well: https://docs.openwebui.com/getting-started/logging

this link now appears to be dead.

**Reply by silentoplayz** [reply]
> > I suppose this may help as well: https://docs.openwebui.com/getting-started/logging
> 
> this link now appears to be dead.

Moved link: https://docs.openwebui.com/getting-started/advanced-topics/logging
